cpp/src/arrow/adapters/orc/util.cc

```diff
@@ -1111,7 +1121,20 @@ Result<std::shared_ptr<DataType>> GetArrowType(const liborc::Type* type) {
     case liborc::CHAR:
       return fixed_size_binary(static_cast<int>(type->getMaximumLength()));
     case liborc::TIMESTAMP:
+      // Values of TIMESTAMP type are stored in the writer timezone in the Orc file.
```

Comment:
I have added extensive comment to explain the issue and suggested usage. Please check again. Thanks @wjones127